sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -724,6 +758,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
 object HiveExternalCatalog {
   val DATASOURCE_PREFIX = "spark.sql.sources."
   val DATASOURCE_PROVIDER = DATASOURCE_PREFIX + "provider"
+  val DATASOURCE_TABLE_LOCATION = DATASOURCE_PREFIX + "tableLocation"
```

Comment:
cc @yhuai we don't have to use `path` here, as it's stored in table properties, not storage properties. When we write the table location, we only write it to table properties, but when we read it, we should try table properties first, then storage properties, for backward compatibility.